src/VisualStudio/Core/Next/Remote/JsonRpcClient.cs

```diff
@@ -127,6 +127,18 @@ private bool ReportUnlessCanceled(Exception ex, CancellationToken cancellationTo
                 return true;
             }
 
+            if (ex is OperationCanceledException && ((IDisposableObservable)_rpc).IsDisposed)
+            {
+                // JsonRpc will throw its own cancellation token if connection it is attached to
+                // is going away. we can get into this sitatuion when VS is shutting down since we dispose
```

Comment:
this seems strange.  can we file a bug on them to not do this?  i'm ok with us being resilient to it.  it just seems wrong that some random API we're calling is throwing an operationcanceled exception when no cancellation is involved with them.

Reply:
Why do you think a `TimeoutException` would be appropriate here? I don't see any timing related exception being thrown.
I haven't heard of a Connection Closed exception. Can you clarify the actual type name you have in mind? I agree that if we're canceling a request because the stream closed that such an exception would sound right.
In the case of `JsonRpc` itself being disposed, what would you have us throw? `ObjectDisposedException`? Even for the requests that came in (but didn't complete) before the object was disposed?

I'm certainly open to discussing these possibilities.

Reply:
@AArnott I mentioned timeout exception specifically for this (http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices.Next/Remote/ServiceHubRemoteHostClient.cs,315)

for this particular case, something like InvocationAbortedException or something like that. and inside of the exception, it has reason why invocation is aborted such as Stream is disposed or JsonRpc is disposed and etc.